File: js/src/Tested/NBodiesSimulation/DynamicNBodiesSimulation.js
import {NBodySimulationMixin} from "./NBodySimulationMixin.js";

export class DynamicNBodiesSimulation
{
    bodies;
    stepCounts;

    constructor(stepCounts)
    {
        Object.assign(DynamicNBodiesSimulation.prototype, NBodySimulationMixin);

        this.stepCounts = stepCounts;
        this.bodies     = [];
        this.init();
    }

    run()
    {
        this.simulate();

        return this.bodies;
    }
}

File: c#/Benchmark/Logic/NBodySimulation/DynamicNBodiesSimulation.cs
namespace Benchmark.Logic.NBodySimulation;

public class DynamicNBodiesSimulation : AbstractNBodySimulation
{
    public DynamicNBodiesSimulation(int stepsCount)
    {
        this.Size = stepsCount;

        this.Init();
    }

    protected override IList<Body> Bodies { get; } = [];

    private void Init()
    {
        for (int i = 0; i < AbstractNBodySimulation.BODY_COUNT; i++) {
            this.Bodies.Add(new());
        }
    }

    public IList<Body> Run()
    {
        this.Simulate();

        return this.Bodies;
    }
}
